SingularityService/src/test/java/com/hubspot/singularity/mesos/SingularityMesosOfferSchedulerTest.java

```diff
@@ -49,68 +62,239 @@ public SingularityMesosOfferSchedulerTest() {
+    longRunningTasksUsage.put(ResourceUsageType.MEMORY_BYTES_USED, 0);
+    assertScoreIs(1, scheduler.score(getOffer(10, 10, SLAVE_ID), taskRequest, Optional.of(getUsage(10, 10, longRunningTasksUsage))));
+
+    // cpu used, no mem used
+    longRunningTasksUsage.put(ResourceUsageType.CPU_USED, 5);
+    longRunningTasksUsage.put(ResourceUsageType.MEMORY_BYTES_USED, 0);
+    assertScoreIs(0.80, scheduler.score(getOffer(5, 10, SLAVE_ID), taskRequest, Optional.of(getUsage(10, 10, longRunningTasksUsage))));
+
+    longRunningTasksUsage.put(ResourceUsageType.CPU_USED, 8);
+    longRunningTasksUsage.put(ResourceUsageType.MEMORY_BYTES_USED, 0);
+    assertScoreIs(0.68, scheduler.score(getOffer(2, 10, SLAVE_ID), taskRequest, Optional.of(getUsage(10, 10, longRunningTasksUsage))));
+
+    // no cpu used, mem used
+    longRunningTasksUsage.put(ResourceUsageType.CPU_USED, 0);
+    longRunningTasksUsage.put(ResourceUsageType.MEMORY_BYTES_USED, mbToBytes(5));
+    assertScoreIs(0.70, scheduler.score(getOffer(10, 5, SLAVE_ID), taskRequest, Optional.of(getUsage(10, 10, longRunningTasksUsage))));
+
+    longRunningTasksUsage.put(ResourceUsageType.CPU_USED, 0);
+    longRunningTasksUsage.put(ResourceUsageType.MEMORY_BYTES_USED, mbToBytes(8));
+    assertScoreIs(0.52, scheduler.score(getOffer(10, 2, SLAVE_ID), taskRequest, Optional.of(getUsage(10, 10, longRunningTasksUsage))));
+
+    // cpu used, mem used
+    longRunningTasksUsage.put(ResourceUsageType.CPU_USED, 5);
+    longRunningTasksUsage.put(ResourceUsageType.MEMORY_BYTES_USED, mbToBytes(5));
+    assertScoreIs(0.50, scheduler.score(getOffer(5, 5, SLAVE_ID), taskRequest, Optional.of(getUsage(10, 10, longRunningTasksUsage))));
+
+    longRunningTasksUsage.put(ResourceUsageType.CPU_USED, 8);
+    longRunningTasksUsage.put(ResourceUsageType.MEMORY_BYTES_USED, mbToBytes(8));
+    assertScoreIs(0.20, scheduler.score(getOffer(2, 2, SLAVE_ID), taskRequest, Optional.of(getUsage(10, 10, longRunningTasksUsage))));
+  }
 
-    // cpu used, mem used --- different request type
-    usagePerRequestType.put(RequestType.SERVICE, ImmutableMap.of(ResourceUsageType.CPU_USED, 0, ResourceUsageType.MEMORY_BYTES_USED, 0));
-    assertScoreIs(0.75, scheduler.score(getOffer(5, 5, slaveId), taskRequest, Optional.of(getUsage(mbToBytes(5), 5, 10, 10, usagePerRequestType, slaveId))));
+  @Test
+  public void itCorrectlyScoresEmptySlaveNonLongRunningTasks() {
+    String slaveId = "slave";
+    Map<ResourceUsageType, Number> longRunningTasksUsage = new HashMap<>();
+    setRequestType(RequestType.ON_DEMAND);
+
+    // new slave (no resources used) -> near perfect score
+    setDeployStatistics(TimeUnit.MINUTES, 5);
+    longRunningTasksUsage.put(ResourceUsageType.CPU_USED, 0);
+    longRunningTasksUsage.put(ResourceUsageType.MEMORY_BYTES_USED, 0);
+    assertScoreIs(.993, scheduler.score(getOffer(10, 10, slaveId), taskRequest, Optional.of(getUsage(10, 10, longRunningTasksUsage))));
```

Comment:
something interesting to note here. A _perfect_ offer for non long running tasks would be a task that runs for 0 seconds on an empty slave. Anything after that will reduce the score. This doesn't hurt us b/c it'll score each offer in the same regard, but it will likely mean that non long running tasks should have a lower minimum score. They are less picky than long running tasks